### PR TITLE
dgit update

### PR DESCRIPTION
**Pull request by bunsenstraat**
test experimental branch for dgit workspaces
manual activation from the plugin manager is needed for testing:
DECENTRALIZED GIT
then add the plugin from https://dgit2-remix.web.app

### CODE REVIEW

File: apps/remix-ide/src/remixAppManager.js [hunk 1]
```diff
@@ -111,6 +111,17 @@ export class RemixAppManager extends PluginManager {
     try {
       const res = await fetch(this.pluginsDirectory)
       plugins = await res.json()
+      plugins.push({
```
Comment (yann300) [inline apps/remix-ide/src/remixAppManager.js:114]: it is a native plugin, would it be possible to  register and activate it the same way we do for native plugin (see app.js for some code examples)

Comment (yann300) [reply apps/remix-ide/src/remixAppManager.js:114]: And maybe you'd like to add it to `requiredPlugins` of remixAppManager.js cause it shouldn't be seen from the plugin manager (or should it?)

Comment (bunsenstraat) [reply apps/remix-ide/src/remixAppManager.js:114]: This was just to make it easy to test. No reason to make it native. It doesn’t actually do a lot. It’s the dgitprovider which is native and does all the work

File: apps/remix-ide/src/app/files/dgitProvider.js [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+      protocol: 'https',
+      ipfsurl: 'https://ipfs.io/ipfs/'
+    }
+  }
+
+  async getGitConfig () {
+    const workspace = await this.call('filePanel', 'getCurrentWorkspace')
+    return {
+      fs: window.remixFileSystem,
+      dir: `.workspaces/${workspace.name}`
```
Comment (yann300) [inline apps/remix-ide/src/app/files/dgitProvider.js:48]: we could do without hardcoding the value. see https://github.com/ethereum/remix-project/blob/master/apps/remix-ide/src/app/files/workspaceFileProvider.js#L10

**Review by yann300 — APPROVED** [review]

File: apps/remix-ide/src/app/files/dgitProvider.js (hunk 2 above)
Comment (bunsenstraat) [reply apps/remix-ide/src/app/files/dgitProvider.js]: ok I handled it.

File: apps/remix-ide/src/remixAppManager.js (hunk 1 above)
Comment (bunsenstraat) [reply apps/remix-ide/src/remixAppManager.js]: it's gone

**Review by LianaHus — APPROVED** [review]

**Pull request closed without merging**